### DISCUSSION TITLE
How to trim all quotes from field?

### DISCUSSION THREAD

**Discussion by mhsimkin** [JoshClose/CsvHelper#2123] [Q&A]
I'm trying to figure out the best way to remove all quotes from a field. Here is a excerpt of the CSV File.

```
"The Way i Used To Be",161,192
"danielle steel",160,374
"stray kids",159,268
"nicholas sparks",159,266
"moleskine",158,525
"frankenstein",158,275
"ana huang",158,238
"""pokemon""",156,344
"""cavallini & co.""",155,727
"NaNa",155,245
"The CRUEL PRINCE",155,205
"a little life",155,189
"the light we carry",155,187
```
I have tired the following:
1) Setting the Mode on the Configuration to Escape and NoEscape,
2) Trying to add a quote character to the whitespace list. (throws error).
3) Using a custom type converter similar to the one here: [https://github.com/JoshClose/CsvHelper/issues/1257]( https://github.com/JoshClose/CsvHelper/issues/1257) .
4) I've tried various combinations of the 1 - 3 and I am not able to get something to work.

#3 will just start to throw TypeConverterException exceptions stating that the conversion cannot be performed.

My type converter is defined as:

```
public class TermStringTypeConverter : DefaultTypeConverter
{
    public override object? ConvertFromString(string? text, IReaderRow row, MemberMapData memberMapData)
    {
        if (string.IsNullOrWhiteSpace(text))
            return null;

        text = text.Trim().Trim('"');
        
        return base.ConvertFromString(text, row, memberMapData);
    }
}
```

I'm using csv.GetRecords<TRecord> and have a class map registered. The configuration is defined as: 

```
        var options = new CsvConfiguration(CultureInfo.InvariantCulture)
        {
            HasHeaderRecord = true,
            ShouldSkipRecord = args => args.Row.Parser.Record!.All(string.IsNullOrWhiteSpace),
            BadDataFound = null,
            MissingFieldFound = null,
            ReadingExceptionOccurred = null
        };
```
and I create the CsvReader this way:

```
                using var csv = new CsvReader(reader, options);
                ObjectResolver.Current = new ObjectResolver(CanResolve, Resolve);
                csv.Context.RegisterClassMap<SearchTermInfoClassMap>();
```

How can I achieve what I need?

I also need to convert the term to be lower case.

Thank you.

-marc

**Comment by ghost** [comment]
Have you tried something like: text.Replace("\\"", string.Empty) ?

**Reply by mhsimkin** [reply]
I don't think string.Replace() will make a difference when the TypeConverterException being thrown is from base.ConvertFromString(), which is after any changes I make to the text.

1) What is the correct approach? Using a custom type converter or setting the correct configuration options, ex: Mode = Escape, Whitespace = [], etc.

2) Why is the type convert throwing the error? I'm registering the type convert as Map(m => m.Term).TypeConverter<TermStringTypeConverter>();

Thanks

-marc
